### PR TITLE
UI enhancements

### PR DESCRIPTION
**Pull request by jeffmahoney**
This PR contains a few UI improvements:

- Use an appropriate shell default based on client OS
- Present a warning when quarantine is not available on the client OS instead of trying to execute anyway

### CODE REVIEW

**Comment by jeffmahoney** [comment]
I haven't investigated how to do that, but as I was writing this up I was thinking about the same thing. Having hard-coded quarantine artifacts isn't very user friendly.

I'm not a Velociraptor expert, but the only mechanisms I'm aware of to associate configuration information with a particular client are labels and client-specific metadata. The latter would probably work as a first-pass, even if it doesn't scale well. A better solution would be to have a way to associate specific metadata with multiple clients, like attaching metadata to labels. Then we'd also have a question of order of application to figure out.

**Comment by scudette** [comment]
As you can see here https://github.com/Velocidex/velociraptor/blob/3952bd47b3df87b0287387514e3360a9a82a9c7e/services/launcher/launcher.go#L202  if the artifact collection request contains `allow_custom_overrides: true` then Velociraptor will search for an artifact that starts with `Custom.` + artifact name first and run that instead.  This is a generic method of overriding various built in artifacts.

We just need to ensure we are setting this flag in the collection request.

**Comment by jeffmahoney** [comment]
I think the request from #1796 was to allow multiple quarantine artifacts, one per group of clients. So we'd need a way to select a specific artifact per client or per group.

**Comment by scudette** [comment]
Looks like we only do it in one spot here https://github.com/Velocidex/velociraptor/blob/085c35ae1485902ddfe977c4683766908010e404/gui/velociraptor/src/components/clients/host-info.js#L407 which allows us to see the report from the custom Generic.Client.Info in case users want to customize that.

We just need to do the same for the quarantine collections.

**Comment by scudette** [comment]
> I think the request from #1796 was to allow multiple quarantine artifacts, one per group of clients. So we'd need a way to select a specific artifact per client or per group.

A custom artifact can already do things based on client types - for example windows ones can call the windows version, linux ones can call the linux ones etc. Basically a `SELECT OS FROM info()` and based on the OS `SELECT * FROM Artifact.Windows.Quarantine() `

Although clients do not have access to labels (since client artifacts run on the client they have no concept of labels) so this is not easy to do.

**Comment by weslambert** [comment]
I think in this case, being able to specify a single custom artifact would work fine, and we can could branch off the primary custom one based off of the OS if we need to do so.

**Comment by scudette** [comment]
I think this PR is great - we should be able to at first have an OS specific artifact if possible - let the user know at least it is not available for the OS in question (until we come up with some sensible defaults in the built in set). Then we can also add a generic one that just branches off as well to support freebsd etc.

File: gui/velociraptor/src/components/clients/shell-viewer.js [hunk 1]
```diff
@@ -618,8 +625,10 @@ class ShellViewer extends Component {
                                     onSelect={(e) => this.setType(e)}
                                     id="bg-nested-dropdown">
                       <Dropdown.Item eventKey="Powershell">Powershell</Dropdown.Item>
-                      <Dropdown.Item eventKey="Cmd">Cmd</Dropdown.Item>
-                      <Dropdown.Item eventKey="Bash">Bash</Dropdown.Item>
+                      { (!this.state.client_os || this.state.client_os === "windows") &&
```
Comment (scudette) [inline gui/velociraptor/src/components/clients/shell-viewer.js:628]: This is very useful :-).

File: gui/velociraptor/src/components/clients/host-info.js [hunk 2]
```diff
@@ -30,25 +31,65 @@ import { runArtifact } from "../flows/utils.js";
...
 
     componentWillUnmount() {
         this.source.cancel();
     }
 
+    checkQuarantineAvailability() {
+        // If the client is running on an OS other than Windows, Linux, or MacOS,
+        // they'll need to define an artifact name to use for it
+        if (this.state.quarantine_artifact === undefined) {
+            this.setState({'quarantine_available' : false});
```
[the remark this answers is not in the document]
Comment (jeffmahoney) [reply gui/velociraptor/src/components/clients/host-info.js:69]: What would a generic quarantine artifact do? IMO it's better to report that the ability to quarantine a host isn't available than to do something less than expected.

[the remark this answers is not in the document]
Comment (scudette) [reply gui/velociraptor/src/components/clients/host-info.js:69]: Yes true - I was more thinking of the case where the OS is not one of the above options and the user did add a Custom.Generic.Qurantine artifact - so in this case there is no Generic.Quarantine but the GUI can detect there is now a custom override and can still quarantine it.

It is not a huge issue as the only other OS we have is freebsd rn and we do not have quarantine for it either.

**Pull request merged**